diff --git a/src/benchmark.cpp b/src/benchmark.cpp
index 1049228..a5cabb0 100644
--- a/src/benchmark.cpp
+++ b/src/benchmark.cpp
@@ -43,25 +43,6 @@ void Benchmark::startFIO(int block_size, int queue_depth, int threads, const QSt
 {
     emit benchmarkStatusUpdate(tr("Preparing..."));
 
-    KAuth::Action dropCacheAction("org.jonmagon.kdiskmark.dropcache");
-    dropCacheAction.setHelperId("org.jonmagon.kdiskmark");
-    QVariantMap args; args["check"] = true; dropCacheAction.setArguments(args);
-    KAuth::ExecuteJob* dropCacheJob = dropCacheAction.execute();
-
-    if (m_settings->shouldFlushCache()) {
-        dropCacheJob->exec();
-
-        if (dropCacheAction.status() != KAuth::Action::AuthorizedStatus) {
-            setRunning(false);
-            return;
-        }
-    }
-    else {
-        dropCacheJob->~ExecuteJob();
-    }
-
-    args["check"] = false; dropCacheAction.setArguments(args);
-
     auto prepareProcess = std::make_shared<QProcess>();
     prepareProcess->start("fio", QStringList()
                           << "--create_only=1"
@@ -104,11 +85,6 @@ void Benchmark::startFIO(int block_size, int queue_depth, int threads, const QSt
 
         emit benchmarkStatusUpdate(statusMessage.arg(index).arg(m_settings->getLoopsCount()));
 
-        if (m_settings->shouldFlushCache()) {
-            dropCacheJob = dropCacheAction.execute();
-            dropCacheJob->exec();
-        }
-
         kill(process->processId(), SIGCONT); // Resume
 
         process->waitForFinished(-1);
diff --git a/src/main.cpp b/src/main.cpp
index d82d018..0d156ca 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -31,6 +31,10 @@ int main(int argc, char *argv[])
         MainWindow w(&settings, &benchmark);
         w.setFixedSize(w.size());
         w.show();
+        
+        QMessageBox::information(0, "KDiskMark",
+                                 "Clearing the pagecache is not supported for snap and will not be performed.\n" \
+                                 "It may cause incorrect performance measurement, namely unreal high speed, while reading.");
 
         return a.exec();
     }
diff --git a/src/mainwindow.ui b/src/mainwindow.ui
index 3e12e47..476d0b6 100644
--- a/src/mainwindow.ui
+++ b/src/mainwindow.ui
@@ -905,7 +905,10 @@
     <bool>true</bool>
    </property>
    <property name="checked">
-    <bool>true</bool>
+    <bool>false</bool>
+   </property>
+   <property name="enabled">
+    <bool>false</bool>
    </property>
    <property name="text">
     <string>Flush Pagecache</string>
